data-prepper-plugins/kafka-source/src/test/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSourceBufferAccumulatorTest.java

```diff
@@ -0,0 +1,152 @@
+	
+	@Mock
+	private PluginMetrics pluginMetrics;
+	
+	@Mock
+	private Buffer<Record<Object>> record;
+
+	@Mock
+	List<Record<Object>> kafkaRecords;
+
+	@BeforeEach
+	void setUp() throws Exception {
+		when(sourceConfig.getTopics()).thenReturn((mockList));
+		when(mockList.get(0)).thenReturn(topicsConfig);
+		when(topicsConfig.getTopic()).thenReturn(topicConfig);
+		when(topicConfig.getSchemaConfig()).thenReturn(schemaConfig);
+		
+		when(topicConfig.getConsumerGroupConfig()).thenReturn(mock(ConsumerConfigs.class));
+		when(topicConfig.getSchemaConfig()).thenReturn(mock(SchemaConfig.class));
+		
+		when(schemaConfig.getRecordType()).thenReturn("string");
+		buffer = new KafkaSourceBufferAccumulator<>(topicConfig, pluginMetrics);
+	}
+
+	@Test
+	void testWriteEventOrStringToBuffer_plaintext_schemaType() throws Exception {
+		TopicConfig topicConfig = new TopicConfig();
+		SchemaConfig schemaConfig = new SchemaConfig();
+		ConsumerConfigs consumerConfigs = new ConsumerConfigs();
+		consumerConfigs.setBufferDefaultTimeout(Duration.ofMillis(100));
+		schemaConfig.setSchemaType("plaintext");
+		topicConfig.setSchemaConfig(schemaConfig);
+		topicConfig.setConsumerGroupConfig(consumerConfigs);
+
+		KafkaSourceBufferAccumulator<String, String> spyBuffer = spy(buffer);
+		doCallRealMethod().when(spyBuffer).getEventRecord("anyString", topicConfig);
+		spyBuffer.getEventRecord("anyString", topicConfig);
+		verify(spyBuffer).getEventRecord("anyString", topicConfig);
+	}
+
+	@Test
+	void testWriteEventOrStringToBuffer_json_schemaType() throws Exception {
+		String json = "{\"writebuffer\":\"true\",\"buffertype\":\"json\"}";
+		TopicConfig topicConfig = new TopicConfig();
+		SchemaConfig schemaConfig = new SchemaConfig();
+		ConsumerConfigs consumerConfigs = new ConsumerConfigs();
+		consumerConfigs.setBufferDefaultTimeout(Duration.ofMillis(100));
+		schemaConfig.setSchemaType("json");
+		topicConfig.setSchemaConfig(schemaConfig);
+		topicConfig.setConsumerGroupConfig(consumerConfigs);
+
+		KafkaSourceBufferAccumulator<String, String> spyBuffer = spy(buffer);
+		doCallRealMethod().when(spyBuffer).getEventRecord(json, topicConfig);
+		spyBuffer.getEventRecord(json, topicConfig);
+		verify(spyBuffer).getEventRecord(json, topicConfig);
+	}
+
+	@Test
+	void testWriteEventOrStringToBuffer_json_schemaType_catch_block() throws Exception {
+		TopicConfig topicConfig = new TopicConfig();
+		SchemaConfig schemaConfig = new SchemaConfig();
+		ConsumerConfigs consumerConfigs = new ConsumerConfigs();
+		consumerConfigs.setBufferDefaultTimeout(Duration.ofMillis(100));
+		schemaConfig.setSchemaType("json");
+		topicConfig.setSchemaConfig(schemaConfig);
+		topicConfig.setConsumerGroupConfig(consumerConfigs);
+
+		KafkaSourceBufferAccumulator<String, String> spyBuffer = spy(buffer);
+		doCallRealMethod().when(spyBuffer).getEventRecord("anyString", topicConfig);
+		spyBuffer.getEventRecord("anyString", topicConfig);
+		verify(spyBuffer).getEventRecord("anyString", topicConfig);
```

Comment:
We should assertions in additions to `verify`. All tests should try to do more assertions.